open-hackathon-server/src/hackathon/views/schema/AdminHostserverResource

```diff
@@ -0,0 +1,143 @@
+{
+  "AdminHostserverResource": {
+    "get": {
+      "input": {
+        "title": "request to get the infomation of a host_server",
+        "description": "args to create new hackathon. Args that not listed here are NOT acceptable.",
+        "type": "object",
+        "properties": {
+        }
+      }
+    },
+
+    "post": {
+      "input": {
+        "title": "request to create a new hackathon",
+        "description": "args to create new hackathon. Args that not listed here are NOT acceptable.",
+        "type": "object",
+        "properties": {
+          "id": {
+            "description": "id of the host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "hackathon_id": {
+            "description": "id of the hackathon",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "vm_name": {
+            "description": "the name of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "public_dns": {
+            "description": "the public_dns of vm/host_server",
+            "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
```

Comment:
suppose `public_dns` should be a dns name like `***.chinacloudapp.cn`. It's not an IP address.